### ISSUE TITLE
Add AskUserQuestion to KNOWN_CLAUDE_TOOLS

### ISSUE THREAD

**Issue by aRustyDev** [carlrannaberg/cclint#2]
## Summary

`AskUserQuestion` is a valid Claude Code tool but is not included in the `KNOWN_CLAUDE_TOOLS` set in `src/lib/schemas.ts`. This causes false-positive "Unknown tool" warnings on any command or agent that includes it in `allowed-tools`.

## Current Behavior

```
.claude/commands/create-agent.md:
  ⚠ Unknown tool: AskUserQuestion
```

This warning appears on every command file that uses `AskUserQuestion` in its `allowed-tools` frontmatter.

## Expected Behavior

`AskUserQuestion` should be recognized as a known tool and not produce a warning.

## Context

`AskUserQuestion` is a built-in Claude Code tool that allows agents and commands to interactively ask the user questions during execution. It's documented in the Claude Code docs and widely used.

Other potentially missing tools worth auditing:
- `EnterPlanMode`
- `AskUserQuestion`
- `TaskCreate` / `TaskUpdate` / `TaskList` / `TaskGet`

## Fix

Add `'AskUserQuestion'` to the `KNOWN_CLAUDE_TOOLS` Set in `src/lib/schemas.ts`.
